CASSANDRA-18124 4.1

smiklosovic opened the pull request:
Thanks for sending a pull request! Here are some tips if you're new here:
 
 * Ensure you have added or run the [appropriate tests](https://cassandra.apache.org/_/development/testing.html) for your PR.
 * Be sure to keep the PR description updated to reflect all changes.
 * Write your PR title to summarize what this PR proposes.
 * If possible, provide a concise example to reproduce the issue for a faster review.
 * Read our [contributor guidelines](https://cassandra.apache.org/_/development/index.html)
 * If you're making a documentation change, see our [guide to documentation contribution](https://cassandra.apache.org/_/development/documentation.html)
 
Commit messages should follow the following format:

```
<One sentence description, usually Jira title or CHANGES.txt summary>

<Optional lengthier description (context on patch)>

patch by <Authors>; reviewed by <Reviewers> for CASSANDRA-#####

Co-authored-by: Name1 <email1>
Co-authored-by: Name2 <email2>

```

The [Cassandra Jira](https://issues.apache.org/jira/projects/CASSANDRA/issues/)

smiklosovic commented on examples/ssl-factory/test/unit/org/apache/cassandra/security/KubernetesSecretsSslContextFactoryTest.java:
```diff
@@ -149,18 +149,6 @@ public void buildTrustManagerFactoryHappyPath() throws IOException
         Assert.assertNotNull(trustManagerFactory);
     }
 
-    @Test(expected = IOException.class)
```
This test was just failing because it was throwing different exception.

Would be great if somebody doublechecked this.

smiklosovic replied:
even though I set keystore_password in `config` in this test, it still says that it is null but it should not ...

maulin-vasavada replied:
I remember I fixed that one. Let me find out where did it go :)

maulin-vasavada replied:
I found it! Since the failing test is about testing for Invalid Keystore file path, I added below lines in the test. Without those two lines, the new code generates `IllegalArgumentException` since Password is mandatory for file based keystores even before getting to read the keystore file path.

`
config.put(KEYSTORE_PASSWORD_ENV_VAR, "MY_KEYSTORE_PASSWORD");
config.put("MY_KEYSTORE_PASSWORD","ThisWontMatter");
`

Commit:
Allow keystore and truststore passwords to be nullable

patch by Maulin Vasavada; reviewed by Stefan Miklosovic and Caleb Rackliffe for CASSANDRA-18124

smiklosovic committed:
DO NOT COMMIT
